app/models/user.rb

```diff
@@ -16,6 +16,7 @@ class User < ApplicationRecord
   has_many :user_providers, dependent: :destroy
   has_many :flags, foreign_key: :flagger_id, dependent: :destroy, inverse_of: :flagger
   has_many :notifications, as: :recipient, dependent: :destroy
+  has_many :announcements, dependent: nil
```

Comment:
This will raise an exception when trying to delete an admin user with announcements because of the foreign key. But I think that's perfectly fine for the moment; a feature that solves the problem of being able to delete admin users, not a bug 😄  Just something we will need to take into consideration with the new admin.

Reply:
> a feature that solves the problem of being able to delete admin users, not a bug smile

Love this line of thinking :laughing: 

I figured nullify defeated the purpose of tracking announcement owners, and destroying announcments when deleting an admin even more so. Although I figure hard deleting an admin account is probably not something that happens that often?

Reply:
Yeah, you're totally correct, it doesn't happen that often. A separate admin_users table in the future so we can soft delete by deactivating an admin instead of deleting will be the ultimate solution for this.